devito/types/dense.py

```diff
@@ -1229,11 +1246,9 @@ def __indices_setup__(cls, **kwargs):
                 time_dim = grid.time_dim if isinstance(save, int) else grid.stepping_dim
             elif not (isinstance(time_dim, Dimension) and time_dim.is_Time):
                 raise TypeError("`time_dim` must be a time dimension")
-
-            dimensions = list(Function.__indices_setup__(**kwargs))
+            dimensions = list(Function.__indices_setup__(**kwargs)[0])
             dimensions.insert(cls._time_position, time_dim)
-
-        return tuple(dimensions)
+        return Function.__indices_setup__(dimensions=dimensions, staggered=staggered)
```

Comment:
hang on, this doesn't make much sense?
the superclass-produced `staggered` information should be reused... while here it's dropped on the floor and the one provided via kwargs is used ?

Reply:
>while here it's dropped on the floor and the one provided via kwargs is used ?

No it's not, this just passes only the needed kwargs to `__indices_setup__` but ti doesn't drop anything. This does what it's supposed to do without modifying kwargs.

Reply:
sorry, was probably drunk when I wrote that